server/docs/getting-started.md

```diff
@@ -2,106 +2,251 @@
+  * The contract generates die rolls for each player
+
+The information above is enough to determine the structure of the **HighRoller** `AppState` and `Action`, both of which are required for Counterfactual Apps. We also note the data structures `ActionType` and `Stage` that the contract developer chose to create:
+
+```solidity
+contract HighRollerApp is CounterfactualApp {
+
+  enum ActionType {
+    START_GAME,
+    COMMIT_TO_HASH,
+    COMMIT_TO_NUM,
+    REVEAL
+  }
+
+  enum Stage {
+    PRE_GAME,
+    COMMITTING_HASH,
+    COMMITTING_NUM,
+    REVEALING,
+    DONE
+  }
+
+  enum Player {
+    FIRST,
+    SECOND
+  }
+
+  struct AppState {
+    address[2] playerAddrs;
+    Stage stage;
+    bytes32 salt;
+    bytes32 commitHash;
+    uint256 playerFirstNumber;
+    uint256 playerSecondNumber;
+  }
+
+  struct Action {
+    ActionType actionType;
+    uint256 number;
+    bytes32 actionHash;
+  }
+```
+
+Because it’s such a simple game (Player1 → Player2 → Player1 → DONE), `getTurnTaker()` and `isStateTerminal()` are easy to implement:
+
+```solidity
+function isStateTerminal(AppState memory state)
+  public
+  pure
+  returns (bool)
+{
+  return state.stage == Stage.DONE;
+}
+
+function getTurnTaker(AppState memory state)
+  public
+  pure
+  returns (Player)
+{
+  return state.stage == Stage.COMMITTING_NUM ? Player.SECOND : Player.FIRST;
+}
+```
+
+Actions for the game are defined (and constrained) in the `applyAction` function:
+
+```solidity
+function applyAction(AppState memory state, Action memory action)
+  public
+  pure
+  returns (bytes memory)
+{
+  AppState memory nextState = state;
+  if (action.actionType == ActionType.START_GAME) {
+    require(
+      state.stage == Stage.PRE_GAME,
+      "Cannot apply START_GAME on PRE_GAME"
+    );
+    nextState.stage = Stage.COMMITTING_HASH;
+  } else if (action.actionType == ActionType.COMMIT_TO_HASH) {
+    require(
+      state.stage == Stage.COMMITTING_HASH,
+      "Cannot apply COMMIT_TO_HASH on COMMITTING_HASH"
+    );
+    nextState.stage = Stage.COMMITTING_NUM;
+
+    nextState.commitHash = action.actionHash;
+  } else if (action.actionType == ActionType.COMMIT_TO_NUM) {
+    require(
+      state.stage == Stage.COMMITTING_NUM,
+      "Cannot apply COMMITTING_NUM on COMMITTING_NUM"
+    );
+    nextState.stage = Stage.REVEALING;
+
+    nextState.playerSecondNumber = action.number;
+  } else if (action.actionType == ActionType.REVEAL) {
+    require(
+    state.stage == Stage.REVEALING,
+    "Cannot apply REVEALING on REVEALING"
+    );
+    nextState.stage = Stage.DONE;
+
+    nextState.playerFirstNumber = action.playerFirstNumber;
+    nextState.salt = action.salt;
+  } else {
+    revert("Invalid action type");
+  }
+  return abi.encode(nextState);
+}
+```
+
+You should also take a look at `highRoller()` which takes in the deterministic “random” seed made from the submitted player numbers and outputs dice roll totals for each player.
```

Comment:
Instead of saying "take a look" it might be a good idea to explain how this Solidity code is use the hash of the product of the numbers (a bytes32 object) and slicing it into four bytes8 objects to represent the dice rolls